core/src/main/java/org/apache/calcite/util/Util.java

```diff
@@ -2166,6 +2167,24 @@ public static <E> boolean isDistinct(List<E> list) {
     return firstDuplicate(list) < 0;
   }
 
+
+  /**
+   * Returns whether the elements of {@code list} are distinct according to a given
+   * case-sensitivity policy.
+   */
+  public static boolean isDistinct(List<String> list, boolean caseSensitive) {
+    if (caseSensitive) {
+      return isDistinct(list);
+    }
+    final Set<String> set = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
if list has 0 or 1 elements you can immediately return true

Reply:
@julianhyde Thank you for taking the time to review, and please review it again.